llvm/lib/Target/RISCV/Disassembler/RISCVDisassembler.cpp

```diff
@@ -453,9 +453,10 @@ static DecodeStatus decodeCLUIImmOperand(MCInst &Inst, uint32_t Imm,
                                          int64_t Address,
                                          const MCDisassembler *Decoder) {
   assert(isUInt<6>(Imm) && "Invalid immediate");
-  if (Imm > 31) {
+  if (Imm == 0)
+    return MCDisassembler::Fail;
+  if (Imm > 31)
```

Comment:
This `Imm > 31` check seems unnecessary. SignExtend64<6> won't do anything to a number less than or equal to 31. But we don't have to change it in this patch.

Reply:
Interesting. A simple enough thing to check and change.